### ISSUE TITLE
Errors in IE11 if imported after react-app-polyfills/ie11

### ISSUE THREAD

**Issue by JustinTRoss** [ljharb/es-abstract#39]
If you import react-app-polyfill/ie11 before importing any shim/polyfill using this lib, you get an error in IE11.

Error: `object doesn't support property or method Symbol(Symbol.iterator)_9.<some-hash>`
Location: GetIntrinsic.js at line 56.
Workaround: import react-app-polyfill/ie11 after importing all shims/polyfills using this lib.

**Comment by ljharb** [comment]
I’m not familiar with react-app-polyfill (i strongly recommend using https://npmjs.com/airbnb-browser-shims and its dependent shims/polyfills). Have you filed this with that project? A polyfill that causes other standards-compliant code to break is a broken polyfill.

**Comment by JustinTRoss** [comment]
I have filed it there as well. See https://github.com/facebook/create-react-app/issues/5615 .

I agree with your assertion. Additionally, I can see the argument being made that a polyfill which is broken by the import of a standards-compliant polyfill is a broken polyfill.

Note - I haven't verified anyone's standards compliance. 😛

**Comment by JustinTRoss** [comment]
I can't find it in ecma standards, but I have noticed the `react-app-polyfills/ie11` polyfill results in `typeof Symbol.iterator === 'object'`. GetIntrinsics does checks on `typeof Symbol.iterator === 'symbol'`, so my guess is that `react-app-polyfills` has deviated a bit.

**Comment by JustinTRoss** [comment]
Nvm. I think that's just how IE11 is showing it after the fact.

**Comment by JustinTRoss** [comment]
Note - The polyfills are actually being imported from `core-js` by `react-app-polyfills/ie11`.

**Comment by JustinTRoss** [comment]
Yeah. The following breaks.

```
import 'core-js/es6/symbol';
import objectValues from 'object.values/polyfill';
```

**Comment by ljharb** [comment]
What would be super nice (but you don't have to feel obligated to provide) is a jsfiddle that can replicate the issue in IE 11 :-)

**Comment by ljharb** [comment]
@JustinTRoss I've been unable to replicate this (what version of `core-js` are you using?), so a jsfiddle would unblock me from fixing this issue.

**Comment by devuxer** [comment]
> What would be super nice (but you don't have to feel obligated to provide) is a jsfiddle that can replicate the issue in IE 11 :-)

Does jsfiddle even work in IE 11? Seems all these services (jsfiddle, codepen, codesandbox) don't support IE 11 anymore. Please correct me if there's a workaround for this, though.

In any case, I'm getting a slightly different IE 11 issue (see [this repo](https://github.com/devuxer/reach-polyfills-issue)) involving [this script](https://polyfill.io/v3/polyfill.js?features=Promise,Object.assign,Object.getOwnPropertyDescriptor) from polyfill.io.

If I look at the flags for this, I see:

 - _ESAbstract.CreateMethodProperty, License: CC0 (required by "Object.assign", "Object.getOwnPropertyDescriptor", "Object.keys", "Function.prototype.bind")
 - _ESAbstract.Get, License: CC0 (required by "Object.assign")
 - _ESAbstract.ToObject, License: CC0 (required by "Object.assign")
 - Object.assign, License: CC0
 - Promise, License: MIT */

So, even though I'm getting a different error (SCRIPT445: Object doesn't support this action) when hitting `Object.getOwnPropertyDescriptor`, I suspect/hope it's related to this issue.

**Comment by ljharb** [comment]
@devuxer i believe it does - but any runnable example would be great.

`Object.getOwnPropertyDescriptor` definitely exists in IE 11; either way, I'd recommend using something predictable and always spec-compliant, instead of polyfill.io.

To reiterate, though - no matter what you're using, if you can provide a runnable reproduction case, I can fix it promptly. I'm still unable to reproduce any issues.

**Comment by devuxer** [comment]
@ljharb,

Can you clarify what you mean by "runnable"? The repo I linked to is runnable, although it does require node to be installed on your machine.

**Comment by ljharb** [comment]
@devuxer i mean a webpage, where i can just load the URL directly in IE 11.

**Comment by devuxer** [comment]
Oh gotcha:

Try this [static react website](https://keen-bardeen-0880d1.netlify.com/) in IE 11:

**Comment by ljharb** [comment]
@devuxer thanks! any chance that could be updated to be unminified, and run in dev mode not prod mode? IE 11 doesn't understand source maps.

**Comment by ljharb** [comment]
aha!!! it seems that IE 11 crashes when you run `Object.getOwnPropertyDescriptor(window.location, 'hash')` (or with any key).

The source code that's crashing seems to start like this:
```js
CreateMethodProperty(Object, 'assign', function assign(target, source) { // eslint-disable-line no-unused-vars
```
This isn't https://npmjs.com/object.assign, and I'm pretty sure it's not core-js's - which means the issue is that this is using a *broken* polyfill for `Object.assign`.

The comments in the file point to https://github.com/financial-times/polyfill-service - thus, I'm closing in favor of https://github.com/Financial-Times/polyfill-service/issues/1898, which is the real culprit.

**Comment by devuxer** [comment]
@ljharb, I really have to try again to use core-js (first time I tried, I couldn't get it to work, but I later realized something was corrupted with my app, so that may have been the cause).

Anyway, I'm not entirely sure my repo covers the OP's problem, so I don't know if this issue should be closed or not, but I'll definitely let you know if the issue disappears with core-js instead of polyfill.io.

**Comment by JustinTRoss** [comment]
Agreed. I'm having trouble seeing any relation between the two issues. Though I no longer have access to the repository in which the issue was discovered, so I'm unable to provide much guidance.

**Comment by ljharb** [comment]
If i can get a repro that includes correct polyfills/shims, I’m more than happy to reopen, since we haven’t yet figured out the issue with react-app-polyfills.

**Comment by devuxer** [comment]
@ljharb,

Just wanted to report back that core-js *did* work! I'm back up and running in IE 11. It sure is slow compared to Chrome, but it works. Guess I'll stick with core-js. I thought polyfill.io was the standard, but this is the second time it's let me down.

**Comment by ljharb** [comment]
I’d never recommend it - always unconditionally apply every polyfill, full stop.
